bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ProfileContextImpl.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.internal;
+
+import java.util.concurrent.ExecutorService;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+
+/**
+ * {@link ProfileContext} implementation.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+@NonNullByDefault
+public class ProfileContextImpl implements ProfileContext {
+
+    private static final String THREAD_POOL_NAME = "profiles";
+    private final Configuration configuration;
+
+    public ProfileContextImpl(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Configuration getConfiguration() {
+        return configuration;
+    }
+
+    @Override
+    public ExecutorService getExecutorService() {
+        return ThreadPoolManager.getPool(THREAD_POOL_NAME);
```

Comment:
When working with this change, i stumbled over this here. 
I am using a scheduledExcecutor in my version. 
Could you call "getScheduledPool" here?
It allows me to do easier scheduling of recurring tasks after initial delay, which is exactly what I need for dimming.

Reply:
Yes, of course (-> #4555) - that was the idea actually...

Reply:
I thought about it yesterday while reviewing your changes, too.
My assumption has been that you would like to prevent the usage of
* `scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit)`
* `scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit)`
in profiles. As it does not make sense to start repetitive stuff.

Reply:
I use the repetitive scheduling to transform long pressed into dimmer increase and decrease commands inside a profile. Should I not be doing that?

Reply:
Yes, I think so too!
Using the scheduler for e.g. repetitively sending commands while a buttons is pressend perfectly makes sense. 
There just shouldn't be any fancy long-running stuff in there, eg. implementing a wake-up light that increases the brightness over 30min is better put into a rule.

Reply:
@kungfoolfighting perhaps your profile should stop increasing if max is reached or decreasing if min is reached. Let's assume an wireless rocker (e.g. EnOcean) and a lost release button telegram. Your scheduled repetitive task should stop some time. :wink:

Reply:
@maggu2810 
There is no way to detect minimum or maximum, since the profile and channel are stateless.
But I have already handled the case of lost release telegrams. I schedule both a repeating task as well as a task that interrupts the repeating task after timeout.

Reply:
> since the profile [...] are stateless

That's not correct - the same profile instance is used for all communications per link. You therefore can keep as much state as you need, if that helps.

Reply:
Sorry bad choice of words. I meant to say that trigger profiles are trigger profiles because the channels they support do not mirror the state of the items to which the are linked. As such I would not want to depend on the item state to make them work. 

But as I said: there isn't a real problem anyways. It all works with the change.